src/error.rs

```diff
@@ -1,72 +1,11 @@
-//! `Ipld` error definitions.
-use alloc::string::String;
-#[cfg(feature = "serde")]
-use alloc::string::ToString;
+//! Common code for `Ipld` errors.
+use crate::ipld::Ipld;
 
-use crate::ipld::{Ipld, IpldIndex};
-pub use anyhow::{Error, Result};
-#[cfg(feature = "std")]
-
-/// Error during Serde operations.
-#[cfg(feature = "serde")]
-#[derive(Clone, Debug)]
-pub struct SerdeError(String);
-
-#[cfg(feature = "serde")]
-impl core::fmt::Display for SerdeError {
-    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        write!(f, "Serde error: {}", self.0)
-    }
-}
-
-#[cfg(feature = "serde")]
-impl serde::de::Error for SerdeError {
-    fn custom<T: core::fmt::Display>(msg: T) -> Self {
-        Self(msg.to_string())
-    }
-}
-
-#[cfg(feature = "serde")]
-impl serde::ser::Error for SerdeError {
-    fn custom<T: core::fmt::Display>(msg: T) -> Self {
-        Self(msg.to_string())
-    }
-}
-
-#[cfg(feature = "serde")]
-impl serde::ser::StdError for SerdeError {}
-
-/// Type error.
-#[derive(Clone, Debug)]
-pub struct TypeError {
-    /// The expected type.
-    pub expected: TypeErrorType,
-    /// The actual type.
-    pub found: TypeErrorType,
-}
-
-impl TypeError {
-    /// Creates a new type error.
-    pub fn new<A: Into<TypeErrorType>, B: Into<TypeErrorType>>(expected: A, found: B) -> Self {
-        Self {
-            expected: expected.into(),
-            found: found.into(),
-        }
-    }
-}
-
-impl core::fmt::Display for TypeError {
-    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        write!(f, "Expected {:?} but found {:?}", self.expected, self.found)
-    }
-}
-
-#[cfg(feature = "std")]
-impl std::error::Error for TypeError {}
-
-/// Type error type.
+/// Error types for the IPLD kinds.
+///
+/// It maps the IPLD kinds into a unity-only enum.
 #[derive(Clone, Debug)]
-pub enum TypeErrorType {
+pub enum KindErrorType {
```

Comment:
Do we need this to be error specific? Or can we just have a general `Kind` enum?

Reply:
That's a good idea!